source/extensions/access_loggers/grpc/config_utils.cc

```diff
@@ -14,9 +14,21 @@ GrpcCommon::GrpcAccessLoggerCacheSharedPtr
 getGrpcAccessLoggerCacheSingleton(Server::Configuration::CommonFactoryContext& context) {
   return context.singletonManager().getTyped<GrpcCommon::GrpcAccessLoggerCacheImpl>(
       SINGLETON_MANAGER_REGISTERED_NAME(grpc_access_logger_cache), [&context] {
+        auto* filter_factory_context =
```

Comment:
This is hacky. I would prefer to move the `getServerFactoryContext()` into the `CommonFactoryContext` interface.

Reply:
See also https://github.com/envoyproxy/envoy/pull/18096 which refactors the factory contexts.

Reply:
Thank you for the comment!
I prefer add getServerFactoryContext() as well. 

I will make the change

Reply:
Perhaps  CommonFactoryContext should not return ServerFactoryContext because ServerFactoryContext is a derive class of the former.

Instead I add the `serverScope()` in CommonFactoryContext. PTAL